crates/oxc_linter/src/rules/typescript/no_extraneous_class.rs

```diff
@@ -5,40 +5,60 @@ use oxc_ast::{
 use oxc_diagnostics::OxcDiagnostic;
 use oxc_macros::declare_oxc_lint;
 use oxc_span::Span;
+use schemars::JsonSchema;
 
 use crate::{AstNode, context::LintContext, rule::Rule};
 
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Clone, JsonSchema)]
+#[serde(renameAll = "camelCase", default)]
 pub struct NoExtraneousClass {
     allow_constructor_only: bool,
     allow_empty: bool,
     allow_static_only: bool,
     allow_with_decorator: bool,
 }
 
+impl Default for NoExtraneousClass {
+    fn default() -> Self {
+        Self {
+            allow_constructor_only: false,
+            allow_empty: false,
+            allow_static_only: false,
+            allow_with_decorator: true,
```

Comment:
:+1: to @Sysix's point, the defaults are all false here:

https://github.com/typescript-eslint/typescript-eslint/blob/a3a157c82a94fb61f4abaf8c4f42d51a7a78fb5a/packages/eslint-plugin/src/rules/no-extraneous-class.ts#L61-L64